pre/file/hash.hpp

```diff
@@ -41,13 +49,31 @@ namespace pre::file {
     }
 
     // compute the hash
-    uint32_t hash[5] = {0};
+    boost::uuids::detail::sha1::digest_type hash = {0};
     char hash_buf[41] = {0};
 
     sha1.get_digest(hash);    
 
-    for (int i = 0; i < 5; i++) {
-      std::sprintf(hash_buf + (i << 3), "%08x", hash[i]);
+    if constexpr (std::size(hash) == 20) {
+      // Newer Boost has digest_type of type "char[20]"
+      for (size_t i = 0; i < std::size(hash); i++) {
+        hash_buf[i * 2 + 0] = int_to_hex[(hash[i] >> 4) & 0xF];
+        hash_buf[i * 2 + 1] = int_to_hex[(hash[i]     ) & 0xF];
+      }
+    } else if constexpr (std::size(hash) == 5) {
+      // Older Boost has digest_type of type "unsigned int[20]"
```

Comment:
`unsigned int[5]` ?

Reply:
also same-ish as above, `unsigned int` isn't guaranteed to be 32bits, so this would required

```cpp
using U = std::make_unsigned_t<decltype(hash[i])>;
    static_assert(sizeof(U) == 4, "Expected 32-bit words in SHA-1 digest");
```

Reply:
Yes, 5, bad copy paste!
As for the `static_assert`, it won't work. Both branches of an `if constexpr` are evaluated and all `static_assert`s are supposed to be valid, even in the non-taken branch. Since we're trying to be compatible with the new version of Boost that is using `unsigned char`, this will trip.

In any case, the UUID library already has `static_assert`s for the size of those types.